### ISSUE TITLE
Test error with new wcslib (7.1)

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#9968]
### Description

When I try to (re-)build astropy with wcslib 7.1, I get an `astropy.wcs.wcs.FITSFixedWarning`, which leads to erroring the test suite.

### Expected behavior

The test should run without errors and pass

### Actual behavior
```
==================================== ERRORS ====================================
_ ERROR collecting .pybuild/cpython3_3.8/build/astropy/nddata/mixins/tests/test_ndarithmetic.py _
astropy/nddata/mixins/tests/test_ndarithmetic.py:190: in <module>
    ???
astropy/wcs/wcs.py:504: in __init__
    self.fix(translate_units=translate_units)
astropy/wcs/wcs.py:679: in fix
    warnings.warn(
E   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Set DATE-REF to '1858-11-17' from MJD-REF'.
_ ERROR collecting .pybuild/cpython3_3.8/build/astropy/nddata/mixins/tests/test_ndarithmetic.py _
astropy/nddata/mixins/tests/test_ndarithmetic.py:190: in <module>
    ???
astropy/wcs/wcs.py:504: in __init__
    self.fix(translate_units=translate_units)
astropy/wcs/wcs.py:679: in fix
    warnings.warn(
E   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Set DATE-REF to '1858-11-17' from MJD-REF'.
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
===================== 30 skipped, 2 error in 25.57 seconds =====================
```

### Steps to Reproduce

1. Install Debian unstable with the latest wcslib-dev (7.1+ds-2)
2. Install astropy from source
3. add
   ```
   [build]
   use_system_libraries = True
   ```
   to `setup.cfg`
4. build astropy and run the test suite

### System Details

 * Linux-5.3.0-2-amd64-x86_64-with-debian-bullseye-sid
 * Python 3.7.6 (default, Jan 19 2020, 22:34:52) 
  [GCC 9.2.1 20200117]
 * Numpy 1.17.4
 * astropy 4.0
 * Scipy 1.3.3

Full build log [here](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=s390x&ver=4.0-3%2Bb1&stamp=1582195366&file=log).

**Comment by nden** [comment]
This is a warning raised in the new wcslib and astropy turns all warnings into errors. The warning is caught in the master branch and perhaps a similar version conditional fix should be implemented in astropy 4.0.1.

**Comment by olebole** [comment]
Can you give me the PR # or the commit id?

**Comment by nden** [comment]
#9829 
https://github.com/astropy/astropy/pull/9829/files#diff-d4c5b9e74c6c26000684075497033504

**Comment by olebole** [comment]
Thank you; I created a patch for Debian from this for 4.0.
Since it seems to be solved with #9829, shall this kept open?

**Comment by nden** [comment]
Let's keep this open for now. We may want to add this warning check to 4.0.1 in case someone installs newer wcslib with astropy 4.0. @bsipocz What do you think about this - since 4.0.0 is a LTS release technically dependencies stay the same. But if someone installs a system wide version of wcslib 7.1 with astropy 4.0 this will be a problem. It probably doesn't hurt to have the check in 4.0.1.

**Comment by bsipocz** [comment]
I feel I don't understand the situation. The issue has been solved for 4.0.1 as the bundled wcslib has been updated to 7.1. How would an updated system wcslib cause the warning for 4.0.1? 
(I think we can't do anything for the case when they have 4.0 and update the system library, any patch to 4.0 will end up in 4.0.1, and that's already been covered. Or I misunderstand something really badly).

**Comment by nden** [comment]
@bsipocz I missed that wcslib in 4.0.1 was updated to v 7.1. It's not a problem then.
